datadog/tests/resource_datadog_application_key_test.go

```diff
@@ -76,13 +76,110 @@ func TestDatadogApplicationKey_import(t *testing.T) {
 	})
 }
 
+func TestDatadogApplicationKey_ServiceAccount(t *testing.T) {
+	if isRecording() || isReplaying() {
+		t.Skip("This test doesn't support recording or replaying")
+	}
+	t.Parallel()
+	resourceName := "datadog_application_key.foo"
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	applicationKeyName := uniqueEntityName(ctx, t)
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: accProviders,
+		CheckDestroy:      testAccCheckDatadogApplicationKeyDestroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogApplicationKeyServiceAccount(applicationKeyName, "foo"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogApplicationKeyExists(accProvider, resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", applicationKeyName),
+					testAccCheckDatadogApplicationKeyValueMatches(accProvider, resourceName),
+				),
+			},
+		},
+	})
+}
+
+func TestDatadogApplicationKey_ServiceAccountUpdateForceNew(t *testing.T) {
+	if isRecording() || isReplaying() {
+		t.Skip("This test doesn't support recording or replaying")
+	}
+	t.Parallel()
+	resourceName := "datadog_application_key.foo"
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	applicationKeyName := uniqueEntityName(ctx, t)
+	accProvider := testAccProvider(t, accProviders)
+	applicationKeyNameUpdate := applicationKeyName + "-2"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: accProviders,
+		CheckDestroy:      testAccCheckDatadogApplicationKeyDestroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogApplicationKeyServiceAccount(applicationKeyName, "foo"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogApplicationKeyExists(accProvider, resourceName),
```

Comment:
Do we need both `testAccCheckDatadogApplicationKeyExists` and `testAccCheckDatadogApplicationKeyValueMatches`? I am wondering whether `testAccCheckDatadogApplicationKeyValueMatches` should be good enough as it satisfies both existence and value match tests and would end up making only 1 GET call to the server.

Reply:
great catch! addressed in bd2a69fe